src/qt/walletcontroller.cpp

```diff
@@ -119,25 +121,11 @@ WalletModel* WalletController::getOrCreateWallet(std::unique_ptr<interfaces::Wal
     connect(wallet_model, &WalletModel::coinsSent, this, &WalletController::coinsSent);
 
     // Notify walletAdded signal on the GUI thread.
-    if (QThread::currentThread() == thread()) {
-        addWallet(wallet_model);
-    } else {
-        // Handler callback runs in a different thread so fix wallet model thread affinity.
-        wallet_model->moveToThread(thread());
-        bool invoked = QMetaObject::invokeMethod(this, "addWallet", Qt::QueuedConnection, Q_ARG(WalletModel*, wallet_model));
-        assert(invoked);
-    }
+    Q_EMIT walletAdded(wallet_model);
```

Comment:
Just to be clear, this would work even without `qRegisterMetaType<WalletModel*>()` added in #16348?

Reply:
Yes.